core/libs/commonwealth/commonwealth/mavlink_comm/MavlinkComm.py

```diff
@@ -93,6 +93,8 @@ async def get_most_recent_vehicle_id(self) -> int:
                     last_update_str = message["status"]["time"]["last_update"]
                     # drop sub-microsecond precision as it is not supported by datetime.fromisoformat
                     last_update_str = re.sub(r"(\.\d{6})\d+", r"\1", last_update_str)
+                    # drop Z as it is not supported by datetime.fromisoformat
+                    last_update_str = last_update_str.rsplit("Z", 1)[0]
```

Comment:
There is a regex right above, we can add an optional 'Z' outside the capturing group:
```Diff
- last_update_str = re.sub(r"(\.\d{6})\d+", r"\1", last_update_str)
+ last_update_str = re.sub(r"(\.\d{6})\d+Z?", r"\1", last_update_str)
```

Reply:
probably, I'll try that. I have a hard time reading regexes in general